packages/x-components/src/components/modals/base-id-modal-close.vue

```diff
@@ -32,36 +22,32 @@
    * @public
    */
   export default defineComponent({
-    components: {
-      NoElement
-    },
+    name: 'BaseIdModalClose',
     props: {
-      /**
-       * The modalId of the modal that will be closed.
-       *
-       * @public
-       */
+      /** The modalId of the modal that will be closed. */
       modalId: {
         type: String,
         required: true
       }
     },
-    setup(props) {
+    setup(props, { slots }) {
       const $x = use$x();
 
       /**
        * Emits the {@link XEventsTypes.UserClickedCloseModal} event with the modalId as payload.
        *
        * @param event - The event triggering the function.
-       * @public
        */
-      const emitCloseModalEvent = ({ target }: Event): void => {
+      function closeModal({ target }: Event) {
         $x.emit('UserClickedCloseModal', props.modalId, { target: target as HTMLElement });
-      };
+      }
 
-      return {
-        emitCloseModalEvent
-      };
+      /* Hack to render through a render-function, the `closingElement` slot or, in its absence,
+       the component itself. It is the alternative for the NoElement antipattern. */
+      const innerProps = { closeModal };
+      return (
+        slots.closingElement ? () => slots.closingElement?.(innerProps)[0] : innerProps
```

Comment:
Previously we were using the slot name as kebab case and I think we should keep doing the same here to not introduce an additional breaking change. So I'd refactor it to `slots['closing-element']`

Reply:
I could have sworn I tried it in kebab-case but for some reason/error, I finally chose this way.
Now, I have tried in kebab-case and everything works, so consider it refactored together with `openingElement`.